support table of contents and purecss table style

Dieken opened the pull request:
Closes #21 
This is to resolve issue https://github.com/aos/temple/issues/21 ，and I also include purecss table style to properly render html tables.

I meant to use two separate branches and two pull requests, but they change same line and introduce conflict,  so I send a single pull request to save your trouble.

Dieken committed:
support table of contents with site or page param "enable_toc"

Dieken committed:
support table style with purecss

aos commented:
Hey @Dieken -- thank you! I will have a look at this PR this weekend

Dieken committed:
make code/kbd/pre/samp html tag use grey background

aos commented:
Hey @Dieken -- sorry I've been very busy. I will look at this tonight !

Dieken commented:
It's fine, take your time :-)

I added another minor enhancement to make some tags have grey background color.

aos commented on layouts/_default/single.html:
```diff
@@ -35,7 +35,10 @@ <h1>{{ if .Draft }}DRAFT: {{end}}{{ .Title }}</h1>
 
       <div class="post-content">
-        {{ .Content }}
+        {{ if or .Params.enable_toc (and (eq .Params.enable_toc nil) .Site.Params.enable_toc) -}}
```
Can you change the parameter to just `toc`?

aos replied:
And can you also document this somewhere in the README? Preferably give a shout to it in the `config` example.

Dieken replied:
Parameter name changed.

I document it in README, not duplicate in `exampleSite/config.toml` because the latter is cleaner and I guess people will prefer README first.  Feel free to ask me update `config.toml` too.

aos reviewed (changes requested):
This is great -- thanks! Just a single request and a heads-up. It turns out the implementation of ToC is somewhat bugged, in that, it always parses the headings into a list whether or not it exists:

![image](https://user-images.githubusercontent.com/25783780/54881862-e63f1e80-4e2a-11e9-9f05-2beb7eceb455.png)

You'll notice above that some are "double" bulleted. And that is because the format of this particular post is something like this:
```markdown
## Subheader with no main header
...
## Another subheader
...
#### Fourth subheader with no subheader one level above
...
```

Looks like [there is a solution here](https://github.com/gohugoio/hugo/issues/1778) that might require JS. I prefer to keep this theme clean with as little possible JS.

Dieken committed:
document feature TOC and bypass rendering empty levels

The customization parameter `enable_toc` is renamed to `toc` as
suggested from @aos.

Dieken commented:
@aos  I pushed a new commit which stole a nice workaround to the bug you found,  thank you very much for your comment!

Dieken commented:
Oops, just realized that workaround isn't perfect, it can eliminate the first empty level, but not intermediate empty levels.

It's enough for me,  I prefer to start sections with "##" and leave "#" to the article title,  and I always keep intermediate levels increasing by one.  A fully solution requires enhancement from Hugo, I hope this little patch is enough for you too :-)

aos commented:
Yeah -- I noticed it's a limitation in the Hugo implementation of table of contents! I think they were planning on eventually fixing that. As long as we documented it here, I'm OK with it. Thank you very much!

aos reviewed (approved):
LGTM
